src/index.js

```diff
@@ -0,0 +1,59 @@
+const AMQP = require('simple-amqplib');
+const CustomMigrationSource = require('./migration-source');
+const debug = require('debug')('amqp-log-to-pg');
+const onMessage = require('./on-message');
+const ow = require('ow');
+
+const validateConfig = config => {
+  ow(config, ow.object.partialShape({
+    amqp: ow.object.exactShape({
+      url: ow.string.url,
+      exchange: ow.string.nonEmpty,
+      queue: ow.object.exactShape({
+        name: ow.string.nonEmpty,
+        routingKey: ow.string.nonEmpty
+      })
+    }),
+    db: ow.string.url,
+    tableName: ow.string.nonEmpty
+  }));
+};
+
+let amqp;
+let knex;
+let tableName;
+async function init (config) {
+  validateConfig(config);
+
+  tableName = config.tableName;
+  knex = require('knex')(config.db);
+  amqp = new AMQP(config.amqp);
+  debug('running migrations...');
+  await knex.migrate.latest({ migrationSource: new CustomMigrationSource(config.tableName) });
```

Comment:
why do we need a custom migration source?

Reply:
@aeh Because the `tableName` is a config option and I couldn't find another way to handle that fact.

Reply:
@timlesallen and importing the config and using `config.tableName` in the migration files like you have also done in this PR doesn't do that?

Reply:
@aeh Now that you can pass the config in, you might not be using `config.js`.